openmc/mesh.py

```diff
@@ -809,3 +809,502 @@ def from_xml_element(cls, elem):
+		cv.check_greater_than('p_upper_ratio', p_upper_ratio, 1.0)
+        self._p_upper_ratio = p_upper_ratio
+		
+	@p_survival_ratio.setter
+    def p_survival_ratio(self, p_survival_ratio):
+        cv.check_type('p_survival_ratio', p_survival_ratio, Real)
+		cv.check_greater_than('p_survival_ratio', p_survival_ratio, 1.0)
+        self._p_survival_ratio = p_survival_ratio
+		
+	@p_max_split.setter
+    def p_max_split(self, p_max_split):
+        cv.check_type('p_max_split', p_max_split, Integral)
+		cv.check_greater_than('p_max_split', p_max_split, 1.0)
+        self._p_max_split = p_max_split
+
+	@p_multiplier.setter
+    def p_multiplier(self, p_multiplier):
+        cv.check_type('p_multiplier', p_multiplier, Real)
+		cv.check_greater_than('p_multiplier', p_multiplier, 0.0)
+        self._p_multiplier = p_multiplier
+
+	@biasing_energy.setter
+	def biasing_energy(self, biasing_energy):
+	    cv.check_type('biasing_energy', biasing_energy, Iterable, Real)
+        self._biasing_energy = biasing_energy
+
+	@origin_probability.setter
+	def origin_probability(self, origin_probability):
+	    cv.check_type('origin_probability', origin_probability, Iterable, Real)
+        self._origin_probability = origin_probability
+		
+	@biasing.setter
+	def biasing(self, biasing):
+	    cv.check_type('biasing', biasing, Iterable, Real)
+        self._biasing = biasing
+	
+    @classmethod
+    def to_xml_element(self):
+        """Return XML representation of the WeightWindowMesh
+
+        Returns
+        -------
+        element : xml.etree.ElementTree.Element
+            XML element containing WeightWindowMesh data
+
+        """
+
+        element = ET.Element("weightwindow")
+		
+        if self._type is not None:
+            subelement = ET.SubElement(element, "type")
+            subelement.text = str(self._type)
+			
+        if self._origin is not None:
+            subelement = ET.SubElement(element, "origin")
+            subelement.text = ' '.join(map(str, self._origin))
+			
+        if self._xmesh is not None:
+            subelement = ET.SubElement(element, "xmesh")
+            subelement.text = ' '.join(map(str, self._xmesh))
+			
+        if self._xints is not None:
+            subelement = ET.SubElement(element, "xints")
+            subelement.text = ' '.join(map(str, self._xints))
+			
+        if self._ymesh is not None:
+            subelement = ET.SubElement(element, "ymesh")
+            subelement.text = ' '.join(map(str, self._ymesh))
+			
+        if self._yints is not None:
+            subelement = ET.SubElement(element, "yints")
+            subelement.text = ' '.join(map(str, self._yints))
+
+        if self._zmesh is not None:
+            subelement = ET.SubElement(element, "zmesh")
+            subelement.text = ' '.join(map(str, self._zmesh))
+			
+        if self._zints is not None:
+            subelement = ET.SubElement(element, "zints")
+            subelement.text = ' '.join(map(str, self._zints))
```

Comment:
These can all be replaced by just storing an instance of `RectilinearMesh` on the `WeightWindowMesh` object, and then relying on its `to_xml_element` method. You'll need a check to see whether the `<mesh>` element has already been written -- you can look here for an example:
https://github.com/openmc-dev/openmc/blob/develop/openmc/settings.py#L1010-L1013

Reply:
Similar to above, these options (origin, xmesh, xints, ...) are needed in the XML file to be read by OpenMC. So this cannot rely on the `to_xml_element` method of `RectilinearMesh` class.